ui/src/components/admin/faqs/FaqGrid.js

```diff
@@ -34,24 +67,201 @@ export default function FaqGrid(props) {
+    }
+  };
+
+  // Functions to handle FAQ reordering
+  useEffect(() => {
+    if (api) {
+      api.setSuppressRowDrag(!isReorderingFaqs);
+      if (isReorderingFaqs) {
+        setReorderFaqColumns();
+        setReorderFaqFilters();
+      } else {
+        setReorderFaqColumns();
+        setReorderFaqFilters();
+      }
+    }
+  }, [isReorderingFaqs, api]);
+
+  // Set the column visibility when reordering FAQs
+  const setReorderFaqColumns = () => {
+    if (colApi) {
+      colApi.setColumnVisible('pageTopic', !isReorderingFaqs);
+      colApi.setColumnVisible('createdBy', !isReorderingFaqs);
+      colApi.setColumnVisible('lastUpdatedBy', !isReorderingFaqs);
+      colApi.setColumnVisible('isPublished', !isReorderingFaqs);
+      colApi.setColumnVisible('isDeleted', !isReorderingFaqs);
+    }
+  };
+
+  // Set the filters when reordering FAQs
+  const setReorderFaqFilters = () => {
+    setShowPublishedOnly(isReorderingFaqs);
+    setPageTopicFilter(isReorderingFaqs ? pageTopicList[0] : '');
   };
 
+  // FAQ Publishing & Reordering Functions
+  // Handles FAQ publishing/unpublishing
+  const publishHandler = async () => {
+    if (api) {
+      const selectedFaqs = api.getSelectedNodes();
+
+      if (selectedFaqs && selectedFaqs.length) {
+        if (toPublish) {
+          setToPublishList(selectedFaqs);
+          selectedFaqs.forEach((faq) => {
+            faq.setDataValue('isPublished', true);
+          });
+          resetHandler();
+          setIsReorderingFaqs(true);
+        } else if (toUnpublish) {
+          setToUnpublishList(selectedFaqs);
+          selectedFaqs.forEach((faq) => {
+            faq.setDataValue('isPublished', false);
+          });
+          resetHandler();
+          setIsReorderingFaqs(true);
+        }
+      } else {
+        toast({
+          description: 'No FAQs selected',
+          status: 'error',
+          duration: 3000,
+        });
+      }
+    }
+  };
+  // Enables FAQ reordering
+  const enableFaqReordering = () => {
+    setShowDeleted(false);
+    setIsReorderingFaqs(true);
+  };
+  // Cancels FAQ publishing/reordering
+  const cancelHandler = () => {
+    if (toPublish && toPublishList && toPublishList.length) {
+      toPublishList.forEach((faq) => {
+        faq.setDataValue('isPublished', false);
+      });
+      setToPublish(false);
+    } else if (toUnpublish && toUnpublishList && toUnpublishList.length) {
+      toUnpublishList.forEach((faq) => {
+        faq.setDataValue('isPublished', true);
+      });
+      setToUnpublish(false);
+    }
+    setIsReorderingFaqs(false);
+  };
+  // Set the order of the FAQs when a row is dragged
+  const onRowDragEnd = (params) => {
+    if (params.api) {
+      params.api.forEachNode((node) => {
+        node.setDataValue('order', node.rowIndex + 1);
+      });
+    }
+  };
+
+  // FAQ Update Functions
+  const saveFaqInfo = async (data) => {
+    return updateHandler(data);
+  };
+
+  const saveFaqReordering = async () => {
```

Comment:
might be better to be in a try catch. right now if there is no `api`, you'll never know if it runs